bytecomp/translcore.ml

```diff
@@ -646,10 +649,120 @@ let rec cut n l =
 
 let try_ids = Hashtbl.create 8
 
-let has_tailcall_attribute e =
-  List.exists (fun ({txt},_) -> txt="tailcall") e.exp_attributes
+let is_inline_attribute = function
```

Comment:
I think the fact that we have 100 lines of code dedicated to attribute handling here is the sign that something is wrong. Could we factorize the logic (there are good patterns implemented there) and move this to a utility module?